### ISSUE TITLE
Default constructor for org.zalando.fahrschein.domain.Partition

### ISSUE THREAD

**Issue by mohamedazouz** [zalando-incubator/fahrschein#69]
I'm facing the following exception while trying to get all partition for event as following:
`nakadiClient.getPartitions(eventName)`

Suggestion: could you please add default constructor or `@ JsonCreator` for the current constructor for `org.zalando.fahrschein.domain.Partition` class

The error as following:

```
com.fasterxml.jackson.databind.JsonMappingException: No suitable constructor found for type [simple type, class org.zalando.fahrschein.domain.Partition]: can not instantiate from JSON object (missing default constructor or creator, or perhaps need to add/enable type information?)
 at [Source: sun.net.www.protocol.http.HttpURLConnection$HttpInputStream@41914209; line: 1, column: 3] (through reference chain: java.util.ArrayList[0])
	at com.fasterxml.jackson.databind.JsonMappingException.from(JsonMappingException.java:148)
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.deserializeFromObjectUsingNonDefault(BeanDeserializerBase.java:1106)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserializeFromObject(BeanDeserializer.java:296)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:133)
	at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize(CollectionDeserializer.java:245)
	at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize(CollectionDeserializer.java:217)
	at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize(CollectionDeserializer.java:25)
	at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:3736)
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:2803)
	at org.zalando.fahrschein.NakadiClient.getPartitions(NakadiClient.java:55)

```

**Comment by mohamedazouz** [comment]
Here's a PR to solve the issue https://github.com/zalando-incubator/fahrschein/pull/70

**Comment by jhorstmann** [comment]
Which version of fahrschein and jackson are you using? I don't experience this exception using the 0.4.x release or master and using jackons 2.7.4. If you are using jackson 2.6.x, it might be that the detection of constructors has improved in the newer version.

It might also be that your `ObjectMapper` is not configured to use `SNAKE_CASE` naming conventions:

    objectMapper.setPropertyNamingStrategy(PropertyNamingStrategy.SNAKE_CASE);

Adding an explicit `JsonCreator` might remove that requirement and would be a nice improvement.

**Comment by mohamedazouz** [comment]
We are using the most using master version for `fahrschein` and our jackson version: 2.4.0,
Actually the error is generating from `fahrschein` client when we are trying to get all partitions for an event by the following line
`final List<Partition> partitions = nakadiClient.getPartitions(eventName);`.
 
Do you have any idea why i'm getting such a problem?

**Comment by jhorstmann** [comment]
Jackson 2.4 is quite old, and not a combination that I have tested before. It might be that you run into more issues with this.

I merged your pull request #70, feel free to open more issues in case there are other incompatibilities.

**Comment by mohamedazouz** [comment]
@jhorstmann Thanks 👍
